plugins/repository-hdfs/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+grant {
+  // used for DomainCombiner
+  permission java.security.SecurityPermission "createAccessControlContext";
+  
+  // set TCCL used for bootstrapping Hadoop Configuration and JAAS
+  permission java.lang.RuntimePermission "setContextClassLoader";
+  
+  // Hadoop UGI
+  permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
```

Comment:
After going back and forth (thanks to Eclipse/Gradle cache) it looks like the permission is indeed need. I have updated the permission list (and their cause) [here](https://gist.github.com/costin/ef6c0914a9bee70b419f). In short, it's triggered by the UGI initialization which uses reflection to instantiate some classes:

    at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:115)
    at org.apache.hadoop.security.Groups.<init>(Groups.java:48)
    at org.apache.hadoop.security.Groups.getUserToGroupsMappingService(Groups.java:140)
    at org.apache.hadoop.security.UserGroupInformation.initialize(UserGroupInformation.java:206)
    at org.apache.hadoop.security.UserGroupInformation.setConfiguration(UserGroupInformation.java:227)